tests/visualization_tests/test_optimization_history.py

```diff
@@ -1,26 +1,85 @@
+from typing import List
+
 import numpy as np
 import pytest
 
 from optuna.study import create_study
 from optuna.testing.objectives import fail_objective
 from optuna.trial import Trial
-from optuna.visualization import plot_optimization_history
+from optuna.visualization._optimization_history import _get_optimization_history_info_list
+from optuna.visualization._optimization_history import _get_optimization_history_plot
+from optuna.visualization._optimization_history import _OptimizationHistoryInfo
+from optuna.visualization._optimization_history import _ValuesInfo
 
 
 def test_target_is_none_and_study_is_multi_obj() -> None:
-
     study = create_study(directions=["minimize", "minimize"])
     with pytest.raises(ValueError):
-        plot_optimization_history(study)
+        _get_optimization_history_info_list(
+            study, target=None, target_name="Objective Value", error_bar=False
+        )
+
+
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+@pytest.mark.parametrize("error_bar", [False, True])
+def test_warn_default_target_name_with_customized_target(direction: str, error_bar: bool) -> None:
+    # Single study.
+    study = create_study(direction=direction)
+    with pytest.warns(UserWarning):
+        _get_optimization_history_info_list(
+            study, target=lambda t: t.number, target_name="Objective Value", error_bar=error_bar
+        )
+
+    # Multiple studies.
+    studies = [create_study(direction=direction) for _ in range(10)]
+    with pytest.warns(UserWarning):
+        _get_optimization_history_info_list(
+            studies, target=lambda t: t.number, target_name="Objective Value", error_bar=error_bar
+        )
+
+
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+@pytest.mark.parametrize("error_bar", [False, True])
+def test_info_with_no_trials(direction: str, error_bar: bool) -> None:
+    # Single study.
+    study = create_study(direction=direction)
+    info_list = _get_optimization_history_info_list(
+        study, target=None, target_name="Objective Value", error_bar=error_bar
+    )
+    assert len(info_list) == 0
```

Comment:
Maybe writing this way is more consistent?
```suggestion
    assert info_list == []
```

Reply:
I changed four parts, including this one.